### DISCUSSION TITLE
Error on optimistic mutation with Houdini integration

### DISCUSSION THREAD

**Discussion by Quart3z** [jycouet/kitql#193] [Q&A]
Hi guys, I've encountered an error when trying out the optimistic mutation with ``Houdini`` integration: 
```
Uncaught (in promise) TypeError: Cannot destructure property 'type' of 'selection[fieldName]' as it is undefined
```
What does it means, or am I doing it wrong?

For reference, here is how my mutation looks like:
```
await GQL_CreateAddress.mutate({
  variables: {
    id: ....,
    address: ...,
    .
    .
  },
  optimisticResponse: {
    createAddress: {
      address: {
        id: ...,
        address: ...,
        .
        .
      }
    },
  },
});
```
It works if I remove the ``optimisticResponse``

Thanks in advance!

**Comment by jycouet** [comment]
Thank you for reporting this 👍
As we want to improve error messages, I link to https://github.com/HoudiniGraphql/houdini/issues/362

I guess that what you pass in `optimisticResponse` is not compatible.
Could you do a working mutation (without `optimisticResponse`) copy the exact response and paste it in `optimisticResponse`?
Do you still have the same issue? Change just 1 string to see the behavior?

If you have a small reproduction somewhere, it would be helpful 👍
Let us know.

**Comment by AlecAivazis** [comment]
Can you share your mutation body? Looking at that error, it seems like your optimistic response doesn't match your mutation body. Could one of those fields in the optimistic response not be present in your mutation?

**Comment by Quart3z** [comment]
@jycouet @AlecAivazis thanks for the responses. I've take a look on the content of my ``optimisticResponse``, indeed it has some mismatches with the mutation body.
